src/wallet/coinselection.cpp

```diff
@@ -418,6 +413,7 @@ void SelectionResult::Clear()
 void SelectionResult::AddInput(const OutputGroup& group)
 {
     util::insert(m_selected_inputs, group.m_outputs);
+    m_use_effective = !group.m_subtract_fee_outputs;
```

Comment:
In cb6507ca5fc2be3d88ad5764185419e0f20a175c:

This seems strange, since `m_use_effective` should never really change. IIRC, `m_subtract_fee_outputs` is known right from the start (`CreateTransactionInternal()`) right? It seems that `SelectionResult::m_use_effective` should be a `const bool` and set in the constructor.

Reply:
My expectation would be to `Assume(group.m_subtract_fee_outputs != m_use_effective)`. Maybe I'm misunderstanding?

Reply:
I did it this way because `SelectionResult`s are constructed by the `SelectCoins*` functions which do not know about `m_subtract_fee_outputs` and didn't feel like adding more parameters to those functions just to have `m_use_effective` be set in the constructor.